packages/compass-app-stores/src/stores/instance-store.ts

```diff
@@ -305,55 +308,68 @@ export function createInstancesStore(
     instancesManager.removeMongoDBInstanceForConnection(connectionInfoId);
   });
 
-  on(connections, 'connected', function (instanceConnectionId: string) {
-    const dataService =
-      connections.getDataServiceForConnection(instanceConnectionId);
-    const connectionString = dataService.getConnectionString();
-    const firstHost = connectionString.hosts[0] || '';
-    const [hostname, port] = firstHost.split(':');
-
-    const initialInstanceProps: Partial<MongoDBInstanceProps> = {
-      _id: firstHost,
-      hostname: hostname,
-      port: port ? +port : undefined,
-      topologyDescription: getTopologyDescription(
-        dataService.getLastSeenTopology()
-      ),
-      preferences,
-    };
-    const instance = instancesManager.createMongoDBInstanceForConnection(
-      instanceConnectionId,
-      initialInstanceProps as MongoDBInstanceProps
-    );
+  on(
+    connections,
+    'connected',
+    function (
+      instanceConnectionId: string,
```

Comment:
We're now tacking on more positional arguments. Wouldn't one object  where you can just pick what you want from it start to make more sense? Or do we have too many things using this and that would involve lots of changes? Or is there some other reason I can't immediately spot why we can't change it?

Reply:
I'll take a closer look at the usage, adding a positional was just the easiest for me, but generally speaking yes, we can probably pack it all into one object (also `instanceConnectionId` is just connectionInfo.id, so I'll just drop it I think)

Reply:
I went back and forth on this and decided not to change this, I started moving them to an object of properties, but then `disconnect` event is different and wanted to keep them consistent, but then `connectionId` is like the only used argument in `disconnect` and having it inside an object was weird, so I was trying different combinations, like maybe keeping `connectionId` as an argument, but moving everything else into an object in a second argument, and this also didn't look right, then I thought I'm overthinking it and it was cleaner to just keep it as is. Hope that's okay